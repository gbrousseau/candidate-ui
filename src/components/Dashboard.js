import React, { Component } from 'react';
import ReactTable from "react-table";
import 'react-table/react-table.css';


class Dasboard extends Component {
  constructor(props) {
    super(props);

  }

  render() {
    const { data, columns, pagination } = this.props;

    if (data !== null) {
      return (
        <div style={eStyle}>
          <ReactTable data={data} columns={columns} showPagination={pagination} minRows={0} className={"-striped react-table"}/>
        </div>
      );
    } else {
      return (
        <div style={eStyle}>
          <ReactTable data={data} columns={columns} showPagination={pagination} minRows={0} />
        </div>
      );
    }
  }
}

export default Dashboard;