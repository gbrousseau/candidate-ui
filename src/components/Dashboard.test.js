import React from 'react';
import { render} from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import Dashboard from './Dashboard';

let container = null;

test('renders work orders list in dashboard', async () => {
  const fakeWorkOrders =
    {
      'data': {
        'workOrders': [
          {
            'id': 'k3qoOGdq0B',
            'rootId': 'h7O9kRkk2A',
            'number': '024',
            'title': 'Measure Strap length',
            'description': 'Test description',
            'priority': 'MEDIUM',
            'status': 'OPEN',
            'location': [
              {
                'id': 'DrwFuwsjt0',
                'name': 'Long Beach Convention Center',
                'address': '300 E Ocean Blvd, Long Beach, CA 90802, USA',
                '__typename': 'Location',
              },
            ],
          }, {
            'id': 'fzXzwxgHMC',
            'rootId': null,
            'number': '036',
            'title': 'gary admin delete test',
            'description': null,
            'priority': 'HIGH',
            'status': 'IN_PROGRESS',
            'location': [
              {
                'id': 'DrwFuwsjt0',
                'name': 'Long Beach Convention Center',
                'address': '300 E Ocean Blvd, Long Beach, CA 90802, USA',
                '__typename': 'Location',
              },
            ]
          }
        ]
      }
    };

  jest.spyOn(global, 'fetch').mockImplementation(() =>
    Promise.resolve({
      json: () => Promise.resolve(fakeWorkOrders),
    }),
  );

  await act(async () => {

    container = render(<Dashboard />);
  });

  expect(container.getByTestId('row0-id').textContent).toBe(fakeWorkOrders.data.workOrders[0].id);
  expect(container.getByTestId('row0-title').textContent).toBe(fakeWorkOrders.data.workOrders[0].title);
  expect(container.getByTestId('row0-status').textContent).toContain(fakeWorkOrders.data.workOrders[0].status);

  expect(container.getByTestId('row1-id').textContent).toBe(fakeWorkOrders.data.workOrders[1].id);
  expect(container.getByTestId('row1-title').textContent).toBe(fakeWorkOrders.data.workOrders[1].title);
  expect(container.getByTestId('row1-status').textContent).toContain(fakeWorkOrders.data.workOrders[1].status);

  global.fetch.mockRestore();
});

test('Matches snapshot', async() => {
  let tree = null;
  const fakeWorkOrders =
    {
      'data': {
        'workOrders': [
          {
            'id': 'k3qoOGdq0B',
            'title': 'Measure Strap length',
            'description': 'Test description',
            'priority': 'MEDIUM',
            'status': 'OPEN',
            "updatedAt": "2020-04-26T18:16:00.399Z",
            "createdAt": "2020-04-26T18:16:00.399Z",
            'location': [
              {
                'id': 'DrwFuwsjt0',
                'name': 'Long Beach Convention Center',
                'address': '300 E Ocean Blvd, Long Beach, CA 90802, USA',
                '__typename': 'Location',
              },
            ],
          }, {
            'id': 'fzXzwxgHMC',
            'rootId': null,
            'number': '036',
            'title': 'gary admin delete test',
            'description': null,
            'priority': 'HIGH',
            'status': 'IN_PROGRESS',
            "updatedAt": "2020-04-02T17:44:25.037Z",
            "createdAt": "2020-04-02T17:00:14.735Z",
            'location': [
              {
                'id': 'DrwFuwsjt0',
                'name': 'Long Beach Convention Center',
                'address': '300 E Ocean Blvd, Long Beach, CA 90802, USA',
                '__typename': 'Location',
              },
            ]
          }
        ]
      }
    };

  jest.spyOn(global, "fetch").mockImplementation(() =>
    Promise.resolve({
      json: () => Promise.resolve(fakeWorkOrders)
    })
  );
  await act(async () => {

    tree = await render(<Dashboard />);
  });

  expect(tree).toMatchSnapshot();
});