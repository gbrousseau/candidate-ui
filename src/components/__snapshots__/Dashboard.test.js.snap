// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="margin: 50px 50px 0px 100px; width: auto; text-align: left;"
      >
        <table
          class="table table-striped"
        >
          <thead
            class="thead-dark"
          >
            <tr>
              <th
                scope="col"
              >
                ID
              </th>
              <th
                scope="col"
              >
                Title
              </th>
              <th
                scope="col"
              >
                Location
              </th>
              <th
                scope="col"
              >
                Created
              </th>
              <th
                scope="col"
              >
                Updated
              </th>
              <th
                scope="col"
              >
                Priority
              </th>
              <th
                scope="col"
              >
                Status
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <th>
                <a
                  data-testid="row0-id"
                  href="/workorder/k3qoOGdq0B"
                  style="color: rgb(255, 59, 48);"
                >
                  k3qoOGdq0B
                </a>
              </th>
              <th
                data-testid="row0-title"
              >
                Measure Strap length
              </th>
              <td>
                Long Beach Convention Center
              </td>
              <td>
                April 26th 2020, 11:16:00 am
              </td>
              <td>
                April 26th 2020, 11:16:00 am
              </td>
              <td>
                MEDIUM
              </td>
              <td
                data-testid="row0-status"
              >
                OPEN
              </td>
            </tr>
            <tr>
              <th>
                <a
                  data-testid="row1-id"
                  href="/workorder/fzXzwxgHMC"
                  style="color: rgb(255, 59, 48);"
                >
                  fzXzwxgHMC
                </a>
              </th>
              <th
                data-testid="row1-title"
              >
                gary admin delete test
              </th>
              <td>
                Long Beach Convention Center
              </td>
              <td>
                April 2nd 2020, 10:44:25 am
              </td>
              <td>
                April 2nd 2020, 10:00:14 am
              </td>
              <td>
                HIGH
              </td>
              <td
                data-testid="row1-status"
              >
                IN_PROGRESS
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="margin: 50px 50px 0px 100px; width: auto; text-align: left;"
    >
      <table
        class="table table-striped"
      >
        <thead
          class="thead-dark"
        >
          <tr>
            <th
              scope="col"
            >
              ID
            </th>
            <th
              scope="col"
            >
              Title
            </th>
            <th
              scope="col"
            >
              Location
            </th>
            <th
              scope="col"
            >
              Created
            </th>
            <th
              scope="col"
            >
              Updated
            </th>
            <th
              scope="col"
            >
              Priority
            </th>
            <th
              scope="col"
            >
              Status
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th>
              <a
                data-testid="row0-id"
                href="/workorder/k3qoOGdq0B"
                style="color: rgb(255, 59, 48);"
              >
                k3qoOGdq0B
              </a>
            </th>
            <th
              data-testid="row0-title"
            >
              Measure Strap length
            </th>
            <td>
              Long Beach Convention Center
            </td>
            <td>
              April 26th 2020, 11:16:00 am
            </td>
            <td>
              April 26th 2020, 11:16:00 am
            </td>
            <td>
              MEDIUM
            </td>
            <td
              data-testid="row0-status"
            >
              OPEN
            </td>
          </tr>
          <tr>
            <th>
              <a
                data-testid="row1-id"
                href="/workorder/fzXzwxgHMC"
                style="color: rgb(255, 59, 48);"
              >
                fzXzwxgHMC
              </a>
            </th>
            <th
              data-testid="row1-title"
            >
              gary admin delete test
            </th>
            <td>
              Long Beach Convention Center
            </td>
            <td>
              April 2nd 2020, 10:44:25 am
            </td>
            <td>
              April 2nd 2020, 10:00:14 am
            </td>
            <td>
              HIGH
            </td>
            <td
              data-testid="row1-status"
            >
              IN_PROGRESS
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
